src/Docs/Resources/current/4-how-to/670-administration-language-switcher.md

```diff
@@ -0,0 +1,256 @@
+
+```twig
+<sw-language-switch v-bind:saveChangesFunction="saveOnLanguageChange"
+                    v-bind:abortChangeFunction="abortOnLanguageChange"
+                    v-on:on-change="loadItem">
+</sw-language-switch>
+``` 
+
+```javascript
+import template from './foobar-detail.html.twig';
+
+Shopware.Component.register('foobar-detail', {
+    template,
+
+    inject: [
+        'repositoryFactory',
+        'context'
+    ],
+
+    data() {
+        return {
+            item: null
+        };
+    },
+
+    created() {
+        this.loadItem();
+    },
+
+    computed: {
+        customEntityRepository() {
+            return this.repositoryFactory.create('foobar_custom_entity');
+        }
+    },
+
+    methods: {
+        loadItem() {
+            return this.customEntityRepository.get(this.$route.params.id, this.context).then(entity => {
+                this.item = entity
+            })
+        },
+
+        abortOnLanguageChange() {
+            return this.customEntityRepository.hasChanges(this.item)
+        },
+
+        saveOnLanguageChange() {
+            return this.customEntityRepository.save(this.item, this.context)
+        }
+    }
+});
+```
+
+It is regarded a good way to communicate the current selected language as different values are displayed than usual.
+For that you can simply use the language info component.
+This component displays a short info text about the current selected language and its impacts on the current entity.
+You can see it regularly above an other content:
+
+```twig
+<sw-page class="foobar-detail">
+    <sw-card-view slot="content"
+                  v-if="item">
+        <sw-language-info v-bind:entityDescription="item.name"></sw-language-info>
+    </sw-card-view>
+</sw-page>
+```
+
+
+## Change language in a create page
+
+You simply don't.
+As previously explain on switching the language every content needs to be reloaded.
+On a creation page you have nothing to reload.
+As it is a common use-case to re-use the editing page for creation you have to make sure that you don't reload the language.
+Simply disable the component using the attribute:
+
+```twig
+<sw-page>
+    <template slot="language-switch">
+        <sw-language-switch disabled="disabled"></sw-language-switch>
+    </template>
+</sw-page>
+```
+
+There is still a pitfall to bypass.
+When you create a new entry using a different language than the default language there is not fallback value displayed as no default value is given.
+To ensure a translation for the fallback language you simply force a change to the default language before editing starts:
+
+```javascript
+Shopware.Component.register('foobar-detail', {
+    created() {
+        if (this.languageStore.getCurrentId() !== this.languageStore.systemLanguageId) {
+            this.languageStore.setCurrentId(this.languageStore.systemLanguageId)
+        }
+    },
+
+    computed: {
+        languageStore() {
+            return Shopware.State.getStore('language')
```

Comment:
`Stores` shouldn't be used anymore. Therefore, you shouldn't provide this.

Reply:
These are my notes from copying mainly from sw-manufacturer-… . Can you provide the new way how to do this? I will update it.

Reply:
Sure!
`<template #language-switch>
                <sw-language-switch @on-change="onChangeLanguage"></sw-language-switch>
            </template>`
plus
`        onChangeLanguage(languageId) {
            this.context.languageId = languageId;
            this.getList();
        }`

To be found in `platform/src/Administration/Resources/administration/src/module/sw-settings-currency/page/sw-settings-currency-list/index.js

Reply:
I will fix this. Thanks

Reply:
Thank you for your contribution :)

Reply:
But how do I reset to system language on creation?